umbral/pre.py

```diff
@@ -1,45 +1,47 @@
-from typing import Tuple, Union, List
+import os
+import typing
+from typing import Dict, List, Optional, Tuple, Union
 
-from bytestring_splitter import BytestringSplitter
 from cryptography.hazmat.backends.openssl import backend
-from cryptography.hazmat.primitives.asymmetric import ec
 from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric.ec import EllipticCurve
 
-from umbral._pre import prove_cfrag_correctness, assess_cfrag_correctness
+from bytestring_splitter import BytestringSplitter
+from umbral._pre import prove_cfrag_correctness
+from umbral.config import default_curve
 from umbral.curvebn import CurveBN
-from umbral.config import default_params, default_curve
 from umbral.dem import UmbralDEM, DEM_KEYSIZE
 from umbral.fragments import KFrag, CapsuleFrag
 from umbral.keys import UmbralPrivateKey, UmbralPublicKey
 from umbral.params import UmbralParameters
 from umbral.point import Point
 from umbral.signing import Signer
 from umbral.utils import poly_eval, lambda_coeff, kdf
-import os
 
 
 class GenericUmbralError(Exception):
     pass
 
 
 class UmbralCorrectnessError(GenericUmbralError):
-    def __init__(self, message, offending_cfrags):
+    def __init__(self, message: str, offending_cfrags: List[CapsuleFrag]) -> None:
         super().__init__(message)
         self.offending_cfrags = offending_cfrags
 
 
 class Capsule(object):
     def __init__(self,
                  params: UmbralParameters,
-                 point_e=None,
-                 point_v=None,
-                 bn_sig=None,
-                 point_e_prime=None,
-                 point_v_prime=None,
-                 point_noninteractive=None,
-                 delegating_pubkey: UmbralPublicKey = None,
-                 receiving_pubkey: UmbralPublicKey = None,
-                 verifying_pubkey=None):
+                 point_e: Optional[Point] = None,
+                 point_v: Optional[Union[int, Point]] = None,
+                 bn_sig: Optional[Union[int, CurveBN]] = None,
+                 point_e_prime: Optional[Point] = None,
+                 point_v_prime: Optional[Union[int, Point]] = None,
+                 point_noninteractive: Optional[Union[int, Point]] = None,
```

Comment:
All `Points` above are just `Optional[Point]`, not `Optional[Union[int, Point]]`. Same for `bn_sig`, but with `CurveBN`.

Reply:
Yeah - @jMyles and were are a bit stumped on how int is passed here during test-runtime.